language/vm/vm_runtime/src/execution_stack.rs

```diff
@@ -88,10 +102,19 @@ where
 
     pub fn peek_at(&self, index: usize) -> Result<&Local, VMInvariantViolation> {
         let size = self.stack.len();
-        Ok(self
-            .stack
-            .get(size - index - 1)
-            .ok_or(VMInvariantViolation::EmptyValueStack)?)
+        if let Some(valid_index) = size.checked_sub(index) {
```

Comment:
```suggestion
        if let Some(valid_index) = size.checked_sub(index).and_then(|idx| idx.checked_sub(1)) {
```
(for the next line too)